docs/Training-Self-Play.md

```diff
@@ -0,0 +1,51 @@
+# Training with Self-Play
+
+ML-Agents provides the functionality to train symmetric, adversarial games with [Self-Play](https://openai.com/blog/competitive-self-play/).
+With self-play, agents learn in adversarial games by competing against past versions of itself. Competing
+against fixed, past versions of the agent's policy provides a more stable, stationary learning environment. This is compared
+to competing against its current self in every episode, which is a constantly changing opponent.
+
+Self-play can be used with our implementations of both [Proximal Policy Optimization (PPO)](Training-PPO.md) and [Soft Actor-Critc (SAC)](Training-SAC.md).
+For more general information on training with ML-Agents, see [Training ML-Agents](Training-ML-Agents.md).
+For more algorithm specific instruction, please see the documentation for [PPO](Training-PPO.md) or [SAC](Training-SAC.md).
+
+Self-play is triggered by including the self-play hyperparameter hierarchy in the trainer configuration file.  Detailed description of the self-play hyperparameters are contained below. Furthermore, to distinguish opposing agents, set the team ID to different integer values in the behavior parameters script on the agent prefab.
+
+![Team ID](images/team_id.png)
+
+See the trainer configuration and agent prefabs for our Tennis example environment for an example.
+
+## Training with self-play
+
+Training with self-play adds additional confounding factors to the usual issues faced by reinforcement learning in general. This guide contains some discussion of the self-play hyperparameters and intuitions for tuning them.
+
+The general tradeoff is between speed/generality/stability of learning
+## Hyperparameters
+
+### Reward Signals
+
+The reward signal should still be used as described in the documentation for the other trainers and [reward signals.](Reward-Signals.md) However, we encourage users to be a bit more conservative when shaping reward functions due to the instability and non-stationarity of learning in adversarial games. Specifically, we encouraging users to begin with the simplest possible reward function (+1 winning, -1 losing) and to allow for more iterations of training to compensate for the sparsity.
+
+### Play against current self ratio
```

Comment:
I think for these hyperparams we should mention when to increase/decrease (e.g. decrease this if the ELO is unstable).